src/inventorymanager.cpp

```diff
@@ -255,82 +342,55 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 	if (try_take_count == 0)
 		try_take_count = list_from->getItem(from_i).count;
 
+	ItemStack src_item = list_from->getItem(from_i);
+	src_item.count = try_take_count;
```

Comment:
`try_take_count` is superfluous.

```suggestion
	if (count > 0)
		src_item.count = count;
```
..and remove the code above.